x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/xcontent/XContentUtilsTests.java

```diff
@@ -62,6 +63,13 @@ public void testAddAuthorizationInfoWithApiKey() throws IOException {
         assertThat(json, equalTo("{\"authorization\":{\"api_key\":{\"id\":\"" + apiKeyId + "\",\"name\":\"" + apiKeyName + "\"}}}"));
     }
 
+    public void testAddAuthorizationInfoWithCloudApiKey() throws IOException {
+        String apiKeyId = randomAlphaOfLength(20);
+        Authentication authentication = AuthenticationTestHelper.randomCloudApiKeyAuthentication(apiKeyId);
+        String json = generateJson(Map.of(AuthenticationField.AUTHENTICATION_KEY, authentication.encode()));
+        assertThat(json, containsString("{\"authorization\":{\"cloud_api_key\":{\"id\":\"" + apiKeyId + "\""));
```

Comment:
Test only asserts `id` field. Using `randomCloudApiKeyAuthentication(apiKeyId)` may not be quite suitable here. Probably it's better to randomize `User` then call `randomCloudApiKeyAuthentication(user)`. This way you can assert all fields as you control user's principal (API key `id`), metadata (`name` and `internal`) and `roles`.